utils_nlp/models/transformers/sequence_classification.py

```diff
@@ -307,20 +249,44 @@ def fit(
             seed (int, optional): Random seed used to improve reproducibility. Defaults to None.
         """
 
+        # get device
+        device, num_gpus = get_device(num_gpus=num_gpus, local_rank=local_rank)
+
+        # move model
+        self.model = move_model_to_device(self.model, device, num_gpus, gpu_ids, local_rank)
```

Comment:
According to https://github.com/huggingface/transformers/blob/715fa638a7601eef9fd0ae9311542e1adc56ef8d/examples/run_glue.py#L152, the multi-gpu training code should be after the amp initialization. Right now, this is done in the child class before calling the `fine_tune` method where the amp initialization happens.